src/main/java/learnositysdk/request/Init.java

```diff
@@ -79,6 +77,12 @@
      */
     private String action = "";
 
+    /**
+     * Currently, Data API requests cannot be altered because the full request
+     * string is not returned by the `generate()` method.
+     */
+    private boolean canAlterRequestData = true;
```

Comment:
Can we introduce and document a new way to hit the Data API so the requests can be altered, and warn on the old-style pattern?

Reply:
The `DataAPI` util allows us to alter the request before signing it, however without a breaking change this sort of thing is going to need to stay in the `Init` class here to at least make the old way of calling it still work properly.

I'm not familiar with ways of warning besides a `System.out.println`, but since the DataAPI util relies on the Init class to reduce code duplication, it would also trigger the same warning. Any suggestions? I guess we could add an extra override for the constructor that accepts a `testing`/`allowAlterations`/`returnFullRequest` boolean flag to the end of the signature that we can then use in the DataAPI util without throwing a warning?

Also looking at this again has reminded me I need to add a test here to ensure this way of calling hasn't changed.

Reply:
At least `System.err.println`. We don't have a logger around?

I agree we can't introduce a breaking change, which is why I'm musing about a different pattern of calling the `DataAPI` class. Maybe not for this PR, though.

Testing FTW!

Reply:
I don't think there is a need for this class member. It's only used once and you could just check this.service against data there instead of storing the value here

Reply:
RESOLVED: We've cleaned this up a little bit with the new approach in the polish commit.